jtec-core/src/main/java/edu/tum/sse/jtec/instrumentation/systemevent/SystemEventInstrumentation.java

```diff
@@ -80,25 +125,24 @@ public void reset() {
      * Adds visitors to the methods for tracing system events.
      */
     private AgentBuilder.Transformer systemEventTransformer() {
-        final String currentPid = ProcessUtils.getCurrentPid();
         return (builder, typeDescription, classLoader, module) ->
-                builder.visit(Advice.withCustomMapping().bind(AdviceOutput.class, outputPath).bind(AdvicePid.class, currentPid)
+                builder.visit(Advice.withCustomMapping()
                                 .to(StringPathInterceptor.class)
                                 .on(ElementMatchers.nameMatches(STRING_PARAMETER_TRACED_METHODS)
                                         .and(ElementMatchers.takesArgument(0, TypeDescription.STRING)))).
-                        visit(Advice.withCustomMapping().bind(AdviceOutput.class, outputPath).bind(AdvicePid.class, currentPid)
+                        visit(Advice.withCustomMapping()
                                 .to(SocketInterceptor.class)
                                 .on(ElementMatchers.nameMatches(SOCKET_ADDRESS_PARAMETER_TRACED_METHODS).and(ElementMatchers.takesArguments(2)))).
-                        visit(Advice.withCustomMapping().bind(AdviceOutput.class, outputPath).bind(AdvicePid.class, currentPid)
+                        visit(Advice.withCustomMapping()
                                 .to(PathInterceptor.class)
                                 .on(ElementMatchers.nameMatches(PATH_PARAMETER_TRACED_METHODS).and(ElementMatchers.takesArgument(0, Path.class)))).
-                        visit(Advice.withCustomMapping().bind(AdviceOutput.class, outputPath).bind(AdvicePid.class, currentPid)
+                        visit(Advice.withCustomMapping()
                                 .to(ClassLoaderInterceptor.class)
-                                .on(ElementMatchers.nameMatches(CLASS_LOADER_TRACED_METHODS))).
-                        visit(Advice.withCustomMapping().bind(AdviceOutput.class, outputPath).bind(AdvicePid.class, currentPid)
+                                .on(ElementMatchers.nameMatches(CLASS_LOADER_TRACED_METHODS).and(ElementMatchers.takesArgument(0, TypeDescription.STRING)))).
```

Comment:
We should check if this is responsible for having `.class` files included in the system events or not!